ext/msgpack/packer.c

```diff
@@ -114,6 +114,40 @@ bool msgpack_packer_try_write_with_ext_type_lookup(msgpack_packer_t* pk, VALUE v
     }
 
     if(ext_flags & MSGPACK_EXT_RECURSIVE) {
+        // HACK: While we call the proc, the current pk->buffer won't be reachable
+        // as it will be stored on the stack.
```

Comment:
What do you mean by not reachable?
Maybe this could work through the mark function of `pk`/`pk->buffer`? And ensuring whatever value wraps the `msgpack_packer_t*` is kept alive long enough.

Reply:
`msgpack_buffer_init` resets `pk->buffer`. However, we store the original state of `pk->buffer` on the stack as `parent_buffer`. Since this does not have a corresponding Ruby object, the original values in `pk->buffer` will not get marked.

Reply:
Thanks for the explanation.

> Since this does not have a corresponding Ruby object, the original values in `pk->buffer` will not get marked.

Could we create a corresponding Ruby object for it? That seems like a nice solution to get it marked properly.